core/xml.js

```diff
@@ -104,7 +108,8 @@ Blockly.Xml.blockToDom = function(block) {
   }
 
   if (block.data) {
-    var dataElement = goog.dom.createDom('data', null, block.data);
+    var dataElement =
+        goog.dom.createDom(goog.dom.TagName.DATA, null, block.data);
```

Comment:
Is there an argument for leaving this untyped?  The XML data tag here is coincidentally named and is unrelated to Firefox's nonstandard HTML data tag (of which I was unaware until just now).
Then the goog.require could be removed from this file.